libcloud/compute/drivers/dimensiondata.py

```diff
@@ -1134,6 +1241,118 @@ def ex_disable_monitoring(self, node):
+        :type   disk_id: ``str``
+
+        :rtype: ``bool``
+        """
+        result = self.connection.request_with_orgId_api_1(
+            'server/%s/disk/%s?delete' %
+            (node.id, disk_id)).object
+        response_code = findtext(result, 'result', GENERAL_NS)
+        return response_code in ['IN_PROGRESS', 'SUCCESS']
+
+    def ex_change_storage_speed(self, node, disk_id, speed):
+        """
+        Remove storage from a node
+
+        :param  node: The server to add storage to
+        :type   node: :class:`Node`
+
+        :param  disk_id: The ID of the disk to remove
+        :type   disk_id: ``str``
+
+        :param  speed: The disk speed type e.g. STANDARD
+        :type   speed: ``str``
+
+        :rtype: ``bool``
+        """
+        create_node = ET.Element('ChangeDiskSpeed', {'xmlns': SERVER_NS})
+        ET.SubElement(create_node, 'speed').text = speed
+        result = self.connection.request_with_orgId_api_1(
+            'server/%s/disk/%s/changeSpeed' %
+            (node.id, disk_id),
+            method='POST',
+            data=ET.tostring(create_node)).object
+        response_code = findtext(result, 'result', GENERAL_NS)
+        return response_code in ['IN_PROGRESS', 'SUCCESS']
+
+    def ex_change_storage_size(self, node, disk_id, size):
+        """
+        Remove storage from a node
+
+        :param  node: The server to add storage to
+        :type   node: :class:`Node`
+
+        :param  disk_id: The ID of the disk to remove
+        :type   disk_id: ``str``
+
+        :param  size: The disk size in GB
+        :type   size: ``int``
+
+        :rtype: ``bool``
+        """
+        create_node = ET.Element('ChangeDiskSize', {'xmlns': SERVER_NS})
+        ET.SubElement(create_node, 'newSizeGb').text = str(size)
+        result = self.connection.request_with_orgId_api_1(
+            'server/%s/disk/%s/changeSize' %
+            (node.id, disk_id),
+            method='POST',
+            data=ET.tostring(create_node)).object
+        response_code = findtext(result, 'result', GENERAL_NS)
+        return response_code in ['IN_PROGRESS', 'SUCCESS']
+
+    def ex_clone_node_to_image(self, node, image_name, image_description=None):
+        """
+        Clone a server into a customer image.
+
+        :param  node: The server to clone
+        :type   node: :class:`Node`
+
+        :param  image_name: The name of the clone image
+        :type   image_name: ``str``
+
+        :param  description: The description of the image
+        :type   description: ``str``
+
+        :rtype: ``bool``
+        """
+        if image_description is None:
+            image_description = ''
+        result = self.connection.request_with_orgId_api_1(
+            'server/%s?clone=%s&desc=%s' %
+            (node.id, image_name, image_description)).object
+        response_code = findtext(result, 'result', GENERAL_NS)
+        return response_code in ['IN_PROGRESS', 'SUCCESS']
```

Comment:
Noticed all those API are async - eventually it might be handy to provide a "sync" interface in libcloud (e.g. this way you could potentially return a `NodeImage` object from this method).